lib/properties.js

```diff
@@ -46,6 +46,28 @@ const merge = (destination, source) => {
 /* eslint-enable eqeqeq */
 
 /**
+ * Recursively traverses a layer namespace and sets the value at the corresponding place in the object
+ * @param {Object} destination The destination of the merge operation.
+ * @param {Array<String>} namespaceArray An array of keys (namespaces) to traverse
+ * @param {Object} source The source that properties are merged from
+ * @return {Object}
+ */
+const recusiveNamespaceMerge = (destination, namespaceArray, source) => {
+  const nextNamespace = namespaceArray.shift();
+  const dest = isPlainObject(destination) ? destination : {};
+
+  if (namespaceArray.length) {
+    dest[nextNamespace] = recusiveNamespaceMerge(dest[nextNamespace] || {}, namespaceArray, source);
```

Comment:
Can we eliminate the `|| {}` here? The `isPlainObject()` call above should handle it.